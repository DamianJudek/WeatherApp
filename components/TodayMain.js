import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 3,
    backgroundColor: 'rgba(30, 30, 30, 0.4)',
    // backgroundColor: 'rgba(0, 0, 0, 0.4)',

    flexDirection: 'column',
    borderRadius: 50,
    marginVertical: 10,
  },
  section: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  subSection: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
  },
  baseText: {
    fontFamily: 'Roboto',
    color: '#fff',
    fontSize: 35,
  },
  bigText: {
    fontSize: 95,
  },
  icon: {
    height: 20,
    width: 20,
    marginRight: 10,
  },
  bigIcon: {
    height: 120,
    width: 120,
  },
});

const TodayMain = (props) => {
  const { humidity, temp, wind, description, icon } = props.weather;
  return (
    <View style={styles.container}>
      <View style={styles.section}>
        <Text style={[styles.baseText, styles.heading]}>{props.city}</Text>
      </View>

      <View style={[styles.section, { flex: 2 }]}>
        <View style={[styles.subSection, { flex: 2 }]}>
          <Text style={[styles.baseText, styles.bigText]}>
            {`${temp}\u2103`}
          </Text>
        </View>
        <View style={[styles.subSection, { justifyContent: 'flex-start' }]}>
          <Image
            style={styles.bigIcon}
            resizeMode="contain"
            source={{
              uri: `http://openweathermap.org/img/wn/${icon}@2x.png`,
            }}
          />
        </View>
      </View>

      <View style={styles.section}>
        <Text style={[styles.baseText, styles.heading]}>{description}</Text>
      </View>
      <View style={styles.section}>
        <View style={styles.subSection}>
          <Image
            style={styles.icon}
            resizeMode="contain"
            source={require('../assets/icons/humidity.png')}
          />
          <Text
            style={[styles.baseText, styles.heading]}>{`${humidity}%`}</Text>
        </View>
        <View style={styles.subSection}>
          <Image
            style={styles.icon}
            resizeMode="contain"
            source={require('../assets/icons/wind.png')}
          />
          <Text style={[styles.baseText, styles.heading]}>{`${wind}m/s`}</Text>
        </View>
      </View>
    </View>
  );
};

export default TodayMain;
