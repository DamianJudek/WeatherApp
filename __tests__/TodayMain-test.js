import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TodayMain from '../components/TodayMain';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders correctly', () => {
  const snap = renderer
    .create(
      <TodayMain
        city={'PoznaÅ„'}
        weather={{
          dayNumber: 5,
          description: 'rain',
          hour: '18:00',
          humidity: 81,
          icon: '10n',
          temp: 18,
          wind: 0.38,
        }}
      />,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
