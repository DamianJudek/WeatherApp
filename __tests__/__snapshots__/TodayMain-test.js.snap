// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(30, 30, 30, 0.4)",
      "borderRadius": 50,
      "flex": 3,
      "flexDirection": "column",
      "marginVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Roboto",
            "fontSize": 35,
          },
          undefined,
        ]
      }
    >
      Poznań
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "flex": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "center",
          },
          Object {
            "flex": 2,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 35,
            },
            Object {
              "fontSize": 95,
            },
          ]
        }
      >
        18℃
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "http://openweathermap.org/img/wn/10n@2x.png",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Roboto",
            "fontSize": 35,
          },
          undefined,
        ]
      }
    >
      rain
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../assets/icons/humidity.png",
          }
        }
        style={
          Object {
            "height": "60%",
            "marginRight": 10,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 35,
            },
            undefined,
          ]
        }
      >
        81%
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../assets/icons/wind.png",
          }
        }
        style={
          Object {
            "height": "60%",
            "marginRight": 10,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Roboto",
              "fontSize": 35,
            },
            undefined,
          ]
        }
      >
        0.38m/s
      </Text>
    </View>
  </View>
</View>
`;
